common/p_interaction.cpp

```diff
@@ -577,6 +577,85 @@ ItemEquipVal P_GivePower(player_t *player, int /*powertype_t*/ power)
 
 #include "v_textcolors.h"
 
+	/*
+ * @brief Player grabbed a resurrect player powerup
+ */
+static void P_ResurrectPlayerPowerUp(player_t* player)
+{
+	// Not lives game? Nothing to do.
+	if (!G_IsLivesGame())
+		return;
+
+	if (!player)
+		return;
+
+	// Grab all the players in game, make a list of their ids, then pick one at random
+	PlayersView ingame = PlayerQuery().notHasLives().execute().players;
+	std::vector<int> ingameplayers;
+	for (const auto& p : ingame)
+	{
+		if (p->id != player->id)
+			ingameplayers.push_back(p->id);
+	}
+
+	if (ingameplayers.empty())
+	{
+		SV_BroadcastPrintf("%s tried to resurrect someone, but everyone is alive!\n",
+		                   player->userinfo.netname.c_str());
+		return;
+	}
+
+	int playerid = M_RandomInt(ingameplayers.size());
+
+	auto it = std::find_if(players.begin(), players.end(),
+	                       [&](player_s& p) { return p.id == playerid; });
+
+	if (it == players.end())
+		return;
+
+	it->lives += 1;
+	it->playerstate = PST_REBORN;
+
+	SV_BroadcastPrintf("%s has brought %s back into the fight!\n",
```

Comment:
Now that all the `Printf` functions are implemented with `{fmt}`, you don't need to call `c_str()`. It will work with a `std::string` (and it does take them by `const&` so there's no extra copies happening).